tests/tracer/test_tracer.py

```diff
@@ -754,8 +757,14 @@ def test_tracer_dogstatsd_url():
         assert str(e) == "Unknown url format for `foo://foobar:12`"
 
 
+@pytest.mark.skip(reason="Fails to Pickle RateLimiter in the Tracer")
```

Comment:
If making it a subprocess tests fails the test, should we just keep it as is instead? Or does that still make this test flaky, so we're silencing it for now?

Reply:
The default start method for the multiprocessing library is fork: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods (on macos AND python>3.8 it is spawn). Forking is not supported by python and can lead to deadlocks in some multithread libraries (like ddtrace). 

Converting this test to a subprocess test solves one of these issues however there are still other issues that cause this test to be flaky. For now I think it's best to skip it